src/app/core/components/sort-button/sort-button.component.ts

```diff
@@ -0,0 +1,27 @@
+import { Component, Input, OnChanges, SimpleChanges } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { MediaSortColumn, MediaSort } from '../../models/table.interface';
+import { NgIconComponent, provideIcons } from '@ng-icons/core';
+import { ionArrowDown, ionArrowUp } from '@ng-icons/ionicons';
+
+@Component({
+  selector: 'app-sort-button',
+  standalone: true,
+  imports: [CommonModule, NgIconComponent],
+  templateUrl: './sort-button.component.html',
+  styleUrls: ['./sort-button.component.scss'],
+  viewProviders: [provideIcons({ ionArrowUp, ionArrowDown })],
+})
+export class SortButtonComponent implements OnChanges {
+  @Input({ required: true }) column!: MediaSortColumn;
+  @Input({ required: true }) sort!: MediaSort;
+  currentDirection: 'asc' | 'desc' | 'none' = 'none';
+
+  ngOnChanges(changes: SimpleChanges) {
+    if (this.column === this.sort.column) {
+      this.currentDirection = this.sort.direction ?? 'none';
+    } else {
+      this.currentDirection = 'none';
+    }
```

Comment:
to be a good component citizen: you should check the changes object for `sort` to be changed otherwise early exit. Feels too much code though.

What about writing a currentDirection getter that derives it's state from the current sort direction. Why did you go that route?